src/Symfony/Component/Mailer/Tests/EventListener/MessengerTransportListenerTest.php

```diff
@@ -44,7 +44,7 @@ public function testMessengerTransportStampViaHeader()
         /* @var TransportNamesStamp $stamp */
         $this->assertInstanceOf(TransportNamesStamp::class, $stamp = $event->getStamps()[0]);
         $this->assertSame(['async'], $stamp->getTransportNames());
-        $this->assertTrue($headers->has('X-Bus-Transport'));
+        $this->assertFalse($message->getHeaders()->has('X-Bus-Transport'));
```

Comment:
The previous assertion relied on the fact that the `Message` constructor clones the passed `Headers` object. I think that's not what we wanna check here, but we rather want to make sure that the listener removes the `X-Bus-Transport` header from the message instead.